sdks/python/apache_beam/transforms/display/display_data.py

```diff
@@ -0,0 +1,133 @@
+"""
+DisplayData, its classes, interfaces and methods.
+"""
+
+from __future__ import absolute_import
+
+import calendar
+from datetime import datetime, timedelta
+import inspect
+import json
+
+__all__ = ['HasDisplayData', 'DisplayDataItem', 'DisplayData']
+
+class HasDisplayData(object):
+  """ Basic interface for elements that contain display data.
+
+  It contains only the display_data method.
+  """
+  def __init__(self, *args, **kwargs):
+    super(HasDisplayData, self).__init__(*args, **kwargs)
+
+  def display_data(self):
+    return {}
+
+  def _namespace(self):
+    return '{}.{}'.format(self.__module__, self.__class__.__name__)
+
+
+class DisplayData(object):
+  def __init__(self, namespace='__main__'):
+    self.namespace = namespace
+    self.items = []
+
+  def populate_items(self, display_data_dict):
+    for key, element in display_data_dict.items():
+      if isinstance(element, HasDisplayData):
+        subcomponent_display_data = DisplayData(element._namespace())
+        subcomponent_display_data.populate_items(element.display_data())
+        self.items += subcomponent_display_data.items
+        continue
+
+      if isinstance(element, dict):
+        self.items.append(
+            DisplayDataItem(self.namespace,
+                            key,
+                            DisplayDataItem._get_value_type(element['value']),
+                            element['value'],
+                            shortValue=element.get('shortValue'),
+                            url=element.get('url'),
+                            label=element.get('label')))
+        continue
+
+      # If it's not a HasDisplayData element,
+      # nor a dictionary, then it's a simple value
+      self.items.append(
+          DisplayDataItem(self.namespace,
+                          key,
+                          DisplayDataItem._get_value_type(element),
+                          element))
+
+  def output(self):
+    return [item.get_dict() for item in self.items]
+
+  @classmethod
+  def create_from(cls, has_display_data):
+    if not isinstance(has_display_data, HasDisplayData):
+      raise ValueError('Element of class {}.{} does not subclass HasDisplayData'
+                       .format(has_display_data.__module__,
+                               has_display_data.__class__.__name__))
+    display_data = cls(has_display_data._namespace())
```

Comment:
I went with this solution because constructors can not be overloaded in Python; so I decided to be 'safe' in case we want to add new ways in which `DisplayData` can be created. (e.g. perhaps `from_dict(dict, namespace)`).

But you are right that the options are misleading. Perhaps we could rename `populate_items` to `_populate_items` to hint its *internal* nature (in any case, neither users nor transform implementers are not supposed to use it).